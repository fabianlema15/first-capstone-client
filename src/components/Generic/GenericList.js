import React from 'react';
import GenericContext from '../../contexts/GenericContext'
import GenericItem from './GenericItem';
import Helper from '../Utils/Helper'

class GenericList extends React.Component{
  static contextType = GenericContext;

  renderObjs() {
    const objList = this.context.objList;
    return Object.keys(objList).map(key =>
      <GenericItem key={objList[key].id} obj={objList[key]} objLabel={Helper.serializeObj(this.context.getObjArray)}/>
    )
  }

  render(){
    return <div>
            {Object.keys(this.context.objList).length>0 ? this.renderObjs() : `You still have no ${this.context.objName}s added`}
      </div>
  }
}

export default GenericList;
