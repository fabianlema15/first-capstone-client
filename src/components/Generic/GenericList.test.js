import React from 'react';
import ReactDOM from 'react-dom';
import GenericList from './GenericList'
import toJson from 'enzyme-to-json'
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('<GenericList />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <GenericList>
      </GenericList>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Renders GenericList as expected', () => {
    const context = { objName: 'Test', objList: {}};
	  const wrapper = shallow(<GenericList/>, { context });
	  expect(toJson(wrapper)).toMatchSnapshot();
	  });

})
