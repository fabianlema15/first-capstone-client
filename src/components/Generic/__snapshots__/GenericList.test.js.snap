// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenericList /> Renders GenericList as expected 1`] = `
<div>
  You still have no undefineds added
</div>
`;
